jb/fftw/bm_time_delay_estimator_many.cpp

```diff
@@ -0,0 +1,206 @@
+class fixture {
+public:
+  fixture(int timeseries)
+      : fixture(nsamples, timeseries) {
+  }
+  fixture(int size, int timeseries)
+      : a_(size)
+      , b_(size)
+      , estimator_(a_, b_)
+      , confidence_(a_)
+      , tde_(a_)
+      , sum2_(a_) {
+    jb::testing::create_triangle_timeseries(size, a_);
+    b_ = jb::testing::delay_timeseries_periodic(
+        a_, expected_delay, sampling_period);
+    sum2_ = jb::testing::sum_square(a_);
+  }
+
+  void run() {
+    estimator_.estimate_delay(confidence_, tde_, a_, b_, sum2_);
+  }
+
+  using tested_type = jb::fftw::time_delay_estimator_many<timeseries_type>;
+  using confidence_type = typename tested_type::confidence_type;
+  using estimated_delay_type = typename tested_type::estimated_delay_type;
+  using sum2_type = typename tested_type::sum2_type;
+
+private:
+  timeseries_type a_;
+  timeseries_type b_;
+  tested_type estimator_;
+  confidence_type confidence_;
+  estimated_delay_type tde_;
+  sum2_type sum2_;
+};
+
+/**
+ * Fixture for boost multi array, only 2 dimensions allowed.
+ *
+ * @tparam T value type stores on the multi array
+ * @tparam A Allocator type to allocate value types
+ */
+template <typename T, typename A>
+class fixture<boost::multi_array<T, 2, A>> {
+public:
+  fixture(int timeseries)
+      : fixture(nsamples, timeseries) {
+  }
+  fixture(int size, int timeseries)
+      : a_(boost::extents[timeseries][size])
+      , b_(boost::extents[timeseries][size])
+      , estimator_(a_, b_)
+      , confidence_(a_)
+      , tde_(a_)
+      , sum2_(a_) {
+    jb::testing::create_triangle_timeseries(size, a_);
+    b_ = jb::testing::delay_timeseries_periodic(
+        a_, expected_delay, sampling_period);
+    sum2_ = jb::testing::sum_square(a_);
+  }
+
+  void run() {
+    estimator_.estimate_delay(confidence_, tde_, a_, b_, sum2_);
+  }
+
+  using timeseries_type = boost::multi_array<T, 2, A>;
+  using tested_type = jb::fftw::time_delay_estimator_many<timeseries_type>;
+  using confidence_type = typename tested_type::confidence_type;
+  using estimated_delay_type = typename tested_type::estimated_delay_type;
+  using sum2_type = typename tested_type::sum2_type;
+
+private:
+  timeseries_type a_;
+  timeseries_type b_;
+  tested_type estimator_;
+  confidence_type confidence_;
+  estimated_delay_type tde_;
+  sum2_type sum2_;
+};
+
+template <typename vector_type>
+void benchmark_test_case(jb::testing::microbenchmark_config const& cfg) {
+  using benchmark = jb::testing::microbenchmark<fixture<vector_type>>;
+  benchmark bm(cfg);
+  auto r = bm.run(cfg.timeseries());
+  bm.typical_output(r);
+}
+
+} // anonymous namespace
+
+int main(int argc, char* argv[]) try {
+  jb::testing::microbenchmark_config cfg;
+  cfg.test_case("float:aligned").process_cmdline(argc, argv);
+
+  std::cout << "Configuration for test\n" << cfg << std::endl;
+
+  // check if it is 2 dimensional
+  bool dim_2 = cfg.timeseries() > 1;
+
+  if (cfg.test_case() == "float:aligned") {
```

Comment:
ACK. I will work on this tomorrow. Now I am going to push the changes to my master to see if everything is fine. There are 2 issues still pending, this one and the benchmark config.
I will fix them tomorrow if you prefer to wait for the whole push